### PR TITLE
 spectacle: use minimal opencv 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

nix store diff-closure:

```
abseil-cpp: 20210324.2 → ∅, -5315.3 KiB
hdf5-cpp: 1.14.3 → ∅, -6032.4 KiB
opencv: -44588.1 KiB
protobuf: 21.12 → ∅, -11380.2 KiB
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/kde/gear/spectacle/default.nix [hunk 1]
```diff
@@ -7,6 +7,13 @@
 mkKdeDerivation {
   pname = "spectacle";
 
-  extraBuildInputs = [qtwayland qtmultimedia opencv];
+  extraBuildInputs = [
+    qtwayland
+    qtmultimedia
+    # https://invent.kde.org/graphics/spectacle/-/blob/master/CMakeLists.txt?ref_type=heads#L83
+    ((opencv.override { runAccuracyTests = false; }).overrideAttrs ({ cmakeFlags, ... }: {
+      cmakeFlags = cmakeFlags ++ [ "-DBUILD_LIST=core,imgproc" ];
```
Comment (K900) [inline pkgs/kde/gear/spectacle/default.nix:15]: Can we maybe get a real parameter on the opencv package for this?

Comment (SuperSandro2000) [reply pkgs/kde/gear/spectacle/default.nix:15]: That's a good idea. Then also the override will be more minimal.

File: pkgs/development/libraries/opencv/4.x.nix [hunk 2]
```diff
@@ -80,6 +80,12 @@
 , doxygen
 , graphviz-nox
 
+, runAccuracyTests ? true
+, runPerformanceTests ? false
+# Modules to enable via BUILD_LIST to build a customized opencv.
+# An empty lists means this setting is ommited which matches upstreams default.
+, enabledModules ? [ ]
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/libraries/opencv/4.x.nix:87]: I had that first but I wanted to convey that this should be a list and switched to that

**Review by Coca162 — APPROVED** [review]
This now builds fine with `nixpkgs.config.cudaSupport = true;` set!

**Comment by LunNova** [comment]
Result of `nixpkgs-review pr 314691` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>kdePackages.spectacle</li>
    <li>kdePackages.spectacle.debug</li>
    <li>kdePackages.spectacle.dev</li>
    <li>kdePackages.spectacle.devtools</li>
  </ul>
</details>

**Comment by hatch01** [comment]
What is this PR waiting for?
Because personally, it prevents me from rebuilding my system, so this is probably the case for others.

**Pull request merged**